src/org/traccar/geofence/GeofenceGeometry.java

```diff
@@ -25,4 +25,33 @@
 
     public abstract void fromWkt(String wkt) throws ParseException;
 
+    public static class Coordinate {
+
+        public static final double DEGREE360 = 360;
+
+        private double lat;
+        private double lon;
+
+        public double getLat() {
+            return lat;
+        }
+
+        public void setLat(double lat) {
+            this.lat = lat;
+        }
+
+        public double getLon() {
+            return lon;
+        }
+
+        // Need not to confuse algorithm by the abrupt reset of longitude
+        public double getLon360() {
+            return lon + DEGREE360;
```

Comment:
So, how do we handle it? It seems like you always add 360 degrees. Won't it result in the same difference?